searchcore/src/vespa/searchcore/proton/feedoperation/documentoperation.h

```diff
@@ -4,35 +4,17 @@
 #include "feedoperation.h"
 #include <vespa/searchcore/proton/common/dbdocumentid.h>
 #include <vespa/document/bucket/bucketid.h>
-#include <persistence/spi/types.h>
-#include <vespa/searchlib/query/base.h>
+#include <vespa/persistence/spi/types.h>
 
 namespace proton {
 
 class DocumentOperation : public FeedOperation
 {
-protected:
-    document::BucketId      _bucketId;
-    storage::spi::Timestamp _timestamp;
-    DbDocumentId            _dbdId;
-    DbDocumentId            _prevDbdId;
-    bool                    _prevMarkedAsRemoved;
-    storage::spi::Timestamp _prevTimestamp;
-    mutable uint32_t        _serializedDocSize; // Set by serialize()/deserialize()
-    uint64_t                _prepare_serial_num;
-
-    DocumentOperation(Type type) noexcept;
-
-    DocumentOperation(Type type, document::BucketId bucketId, storage::spi::Timestamp timestamp) noexcept;
-
-    void assertValidBucketId(const document::DocumentId &docId) const;
-    void assertValidBucketId(const document::GlobalId &docId) const;
-    vespalib::string docArgsToString() const;
-
 public:
+    using Timestamp = uint64_t;
```

Comment:
Consider a comment describing timer resolution, i.e.g if it is microseconds, nanoseconds, or dependent on what resolution is used by std::chrono::system_clock.